### PR TITLE
Refactor rules about when/how to display item-level request links into RequestLink class

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

File: app/models/request_link.rb [hunk 1]
```diff
@@ -17,16 +19,31 @@ def self.for(document:, library:, location:, items: [])
...
     ((in_enabled_location? && any_items_circulate?) || mediated_pageable?) &&
       !all_in_disabled_current_location? &&
       !available_via_temporary_access?
       # Check for real barcodes?  Only for items that are not on-order?
       # Array of procs / methods to be sent configurable on a library basis.
   end
 
+  NEVER = false
+  ALWAYS = true
+  DEPENDS_ON_AVAILABILITY = 2
```
Comment (mjgiarlo) [inline app/models/request_link.rb:32]: 👁️ `2`? What's the significance?

Comment (mjgiarlo) [reply app/models/request_link.rb:32]: Nvm, now I see this:

```
location.request_link&.show_item_level_request_link?(item) == RequestLink::DEPENDS_ON_AVAILABILITY
```

Comment (cbeer) [reply app/models/request_link.rb:32]: Yeah. I don't love it, but 🤷‍♂️

**Review by mjgiarlo — APPROVED** [review]
LGTM

**Review by camillevilla — APPROVED** [review]

**Pull request merged**